WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShippingAddresses/WooShippingEditAddressViewModel.swift

```diff
@@ -369,6 +416,38 @@ extension WooShippingEditAddressViewModel {
             return phoneDigits.count == 10
         }
     }
+
+    /// Whether the phone number is required.
+    ///
+    private func phoneNumberRequired(for country: String?, and state: String?) -> Bool {
+        Self.phoneNumberRequired(addressType: addressType,
+                                    selectedCountryCode: country,
+                                    selectedState: state,
+                                    originCountryCode: originCountryCode,
+                                    originStateCode: originStateCode)
+    }
+
+    /// Whether the phone number is required.
+    /// - Parameters:
+    ///   - addressType: Type of address being edited.
+    ///   - selectedCountryCode: Country code of the selected country for the address being edited.
+    ///   - selectedState: Selected state for the address being edited.
+    ///   - originCountryCode: Country code of the origin address.
+    ///   - originStateCode: State code of the origin address.
+    private static func phoneNumberRequired(addressType: AddressType,
+                                            selectedCountryCode: String?,
+                                            selectedState: String?,
+                                            originCountryCode: String?,
+                                            originStateCode: String?) -> Bool {
+        if addressType == .origin {
+            return true
+        } else {
```

Comment:
Suggestion: We could use a `guard` to exit early to increase readability.

Reply:
Good thought! I considered this a bit more and changed it to a `switch` statement in f344e28, since the logic is really a different case for each address type (origin/destination).